libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java

```diff
@@ -617,17 +613,20 @@ public String getNewAuthToken() {
                 gettingAuthToken = true;
             }
 
-            // Invalidate current auth token
-            clientManager.invalidateToken(lastNewAuthToken);
+            // Invalidate current auth token.
+            final String cachedAuthToken = clientManager.accountManager.peekAuthToken(acc, AccountManager.KEY_AUTHTOKEN);
```

Comment:
This is important, because unless we invalidate the correct token, the subsequent call to ```getAuthToken``` won't go to our implementation - it will go directly to ```AccountManager``` cache instead.